metrics/root.go

```diff
@@ -71,6 +74,39 @@ func (m *Metrics) MustStart() {
 	}
 }
 
+// Push pushes registered metrics to a Prometheus push gateway.
+func (m *Metrics) Push() error {
+	registry := prometheus.NewRegistry()
+	m.Registry.Each(func(name string, metric interface{}) {
+		name = strings.Replace(strings.ToLower(name), ".", "_", -1)
+
+		switch metric := metric.(type) {
+		case *metrics.StandardGauge:
+			g := prometheus.NewGauge(prometheus.GaugeOpts{Name: name})
+			g.Set(float64(metric.Value()))
+			registry.Register(g)
+
+		case *metrics.StandardGaugeFloat64:
+			g := prometheus.NewGauge(prometheus.GaugeOpts{Name: name})
+			g.Set(float64(metric.Value()))
+			registry.Register(g)
+
+		case *metrics.StandardCounter:
+			c := prometheus.NewCounter(prometheus.CounterOpts{Name: name})
+			c.Add(float64(metric.Count()))
+			registry.Register(c)
+		}
+	})
+
+	for _, c := range m.config {
+		if p, ok := c.(*PromPushGWConfiguration); ok {
+			return p.pusher.Gatherer(registry).Push()
```

Comment:
Should we also support Add() ? https://godoc.org/github.com/prometheus/client_golang/prometheus/push#Pusher.Add
Maybe a parameter (add bool or something like that) so we can choose what to do?

Reply:
Why not, but in our use cases we'll deal with CLI commands that are ran one-shot so it doesn't really make any difference in this context IMO.